### PR TITLE
added page for list assigned and followed tasks

### PR DESCRIPTION
**Pull request by Kasvit**
**STATUS**
READY

**DESCRIPTION**

**CHECK LIST**
- [x] specs added
- [x] specs passed
- [ ] code coverage >90%
- [ ] DB indexes added if needed
- [ ] seed.rb file is updated if needed
- [x] PR is reviewed manually again (to make sure you have 100% ready code)
- [x] PR doesn't have conflicts with main branch
- [ ] I've checked new feature as logged in and logged out user if needed
- [x] "let!" is used in specs

### CODE REVIEW

**Comment by Kasvit** [comment]
image
image

File: app/models/task.rb [hunk 1]
```diff
@@ -52,6 +52,9 @@ class Task < ApplicationRecord
                                    .where('users.id = ? AND tasks.title ~* ?', user_id, "\\m#{search}")
                                    .limit(10) }
 
+  scope :assigned, -> (user) { where(assignee_id: user.id)}
```
Comment (igorkasyanchuk) [inline app/models/task.rb]: rename to "by_assignee(..)"

File: app/models/task.rb [hunk 2]
```diff
@@ -52,6 +52,9 @@ class Task < ApplicationRecord
...
+  scope :current_workspace, -> (workspace) { where(project_id: workspace.projects.ids)}
```
Comment (igorkasyanchuk) [inline app/models/task.rb:54]: where(project_id: workspace.project_ids)

Comment (igorkasyanchuk) [reply app/models/task.rb:54]: ok actually I think it's ok, maybe its doing the same

File: app/views/account/workspaces/list.html.slim [hunk 3]
```diff
@@ -0,0 +1,38 @@
+- title "#{@workspace.name}: My tasks"
+.container
+  .level
+    .level-left
+      .title
+        = link_to @workspace.name, account_workspace_path(@workspace)
+  .container
+    - if @workspace.tasks.assigned(current_user).any? || current_user.tasks.current_workspace(@workspace).any?
```
Comment (igorkasyanchuk) [inline app/views/account/workspaces/list.html.slim]: create instance method "my_tasks" as union (ask Tanya) which returns one collection of all tasks

File: config/routes.rb [hunk 4]
```diff
@@ -14,6 +14,7 @@
     resources :workspaces do
       resources :members, only: [:new, :create, :destroy]
       resources :projects, except: [:index]
+      get :list, on: :member
```
Comment (igorkasyanchuk) [inline config/routes.rb]:
```
member do
   get :list
end
```

always use this way to build routes

**Review by igorkasyanchuk — APPROVED** [review]

**Review by igorkasyanchuk — APPROVED** [review]

**Pull request merged**